### ISSUE TITLE
Version 1.x.x introduced an bug where `breaking` changes are display, but there are none

### ISSUE THREAD

**Issue by mima0815** [udamir/api-smart-diff#6]
Using the latest version (`v1.0.1`) show breaking changes on parameters but there are none.

Compare the given spec with itself. The expected result is no changes, but it shows a breaking change on the referenced parameter. Your example [application](https://udamir.github.io/api-smart-diff/) shows the same wrong result. Just copy paste the yaml spec in your application and diff it.

Spec:

```yaml
openapi: 3.0.0
info:
  version: 2.8.0
  title: test api

paths:
  /test/endpoint:
    get:
      parameters:
        - $ref: "#/components/parameters/name"
      responses:
        "200":
          content:
            application/json:
              schema:
                type: object
                properties:
                  data:
                    type: string
components:
  parameters:
    name:
      name: name
      in: query
      schema:
        type: string
      required: true

```

Example code:

```js
import { apiCompare } from "api-smart-diff";
import { readFile } from "fs/promises";

const specText = await readFile("spec.json", "utf-8");
const specObj = JSON.parse(specText);

const { diffs } = apiCompare(specObj, specObj);

console.log(diffs);

```

Output:

```js
[
  {
    path: [ 'paths', '/test/endpoint', 'get', 'parameters', 0 ],
    before: { '$ref': '#/components/parameters/name' },
    action: 'remove',
    type: 'breaking',
    description: '[Removed] required query parameter `name`'
  },
  {
    path: [ 'paths', '/test/endpoint', 'get', 'parameters', 1 ],
    after: { '$ref': '#/components/parameters/name' },
    action: 'add',
    type: 'non-breaking',
    description: '[Added] required query parameter `name`'
  }
]
```

**Comment by udamir** [comment]
fixed in api-smart-diff@1.0.2

**Comment by mima0815** [comment]
thanks for the fix!
